custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go

```diff
@@ -27,53 +28,100 @@ func newClient(translator *translator.Translator) *stackdriverCoreClient {
 	return &stackdriverCoreClient{translator}
 }
 
-// TODO(holubowicz): handle a case when len(resourceNames) > oneOfMax
-func (p *stackdriverCoreClient) getContainerCPU(podsNames []string) (map[string]map[string]resource.Quantity, map[string]api.TimeInfo, error) {
-	stackdriverRequest, err := p.translator.GetSDReqForContainersWithNames(podsNames, containerCPUMetricName, metricKindCPU, labels.Everything(), translator.AllNamespaces)
-	if err != nil {
-		return nil, nil, err
+func (p *stackdriverCoreClient) getPodMetric(podsNames []string, metricName string, metricKind string, labels labels.Selector) (map[string]map[string]resource.Quantity, map[string]api.TimeInfo, error) {
+	numOfRequests := (len(podsNames) + translator.MaxNumOfArgsInOneOfFilter - 1) / translator.MaxNumOfArgsInOneOfFilter // ceil
+
+	responseChannel := make(chan *stackdriver.ListTimeSeriesResponse, numOfRequests)
+	errChannel := make(chan error, numOfRequests)
+	defer close(responseChannel)
+	defer close(errChannel)
+
+	for i := 0; i < numOfRequests; i++ {
+		segmentBeg := i * translator.MaxNumOfArgsInOneOfFilter
+		segmentEnd := min((i+1)*translator.MaxNumOfArgsInOneOfFilter, len(podsNames))
+		stackdriverRequest, err := p.translator.GetSDReqForContainersWithNames(podsNames[segmentBeg:segmentEnd], metricName, metricKind, labels, translator.AllNamespaces)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		go func(stackdriverRequest *stackdriver.ProjectsTimeSeriesListCall) {
```

Comment:
I think paralyzing requests is a good end goal, but it can make code pretty complicated and hard to spot bugs. Could we start synchronous implementation ?

Reply:
Fine, we can start with adding synchronous version